small-angle-demo/src/main.js

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import MathJax from 'react-mathjax2';
+import VisualDemo from './VisualDemo';
+import {forceFloat} from './utils';
+
+class SmallAngleDemo extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            distance: 40,
+            diameter: 2
+        };
+        this.handleInputChange = this.handleInputChange.bind(this);
+    }
+    render() {
+        const result = Math.round(
+            206265 * (this.state.diameter / this.state.distance) * 10)
+              / 10;
+        const tex = `\\alpha = 206{,}265 \\times ` +
+              `{\\text{linear diameter} \\over \\text{distance}} ` +
+              `= ${result} \\text{arcsec}`;
+        return <div>
+            <MathJax.Context
+        script="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-MML-AM_CHTML"
+        input="tex"
+        options={{
+            showProcessingMessages: false,
+            messageStyle: 'none'
+        }}
+            >
+            <div>
+            <MathJax.Node>{tex}</MathJax.Node>
+            </div>
+            </MathJax.Context>
+
+            <div>
+            <VisualDemo
+        distance={this.state.distance}
+        diameter={this.state.diameter} />
+            </div>
+
+            <form className="form-inline">
+            <div>
+            <label htmlFor="distRange">Distance:</label>
+            <input type="number" size="4"
+        className="form-control form-control-sm"
+        step="0.1" name="distance"
+        value={this.state.distance}
+        onChange={this.handleInputChange} /> units
+            <input name="distance" id="distRange"
+        className="form-control form-control-sm custom-range ml-2"
+        value={this.state.distance}
+        onChange={this.handleInputChange}
+        type="range" step="0.1"
+        min="20" max="60" />
+            </div>
+
+            <div className="ml-4">
+            <label htmlFor="diamRange">Diameter:</label>
+            <input type="number" size="4"
+        className="form-control form-control-sm"
+        step="0.1" name="diameter"
+        value={this.state.diameter}
+        onChange={this.handleInputChange} /> units
+            <input name="diameter" id="diamRange"
+        className="form-control form-control-sm custom-range ml-2"
+        value={this.state.diameter}
+        onChange={this.handleInputChange}
+        type="range" step="0.1"
+        min="1" max="3" />
+            </div>
+            </form>
+            </div>;
```

Comment:
Sorry about this mis-formatted JSX/html. It works it just looks awkward. This is the way my emacs jsx plugin formats it - I need to fix that up somehow or try something else.